src/tactic/ui/startup/security_wdg.py

```diff
@@ -412,18 +432,59 @@ def execute(my):
             # make sure the group doesn't already exist
             #login_group = LoginGroup.get_by_login_group(new_group)
 
+            project_included = False
+            user_defined_project_code = ""
+            original_new_group = new_group
+            is_project_specific = False
+            if my.kwargs.get("checked") in [True, "true", "True"]:
+                is_project_specific = True
+
             if new_group.find("/") == -1:
                 title = new_group
                 new_group = "%s/%s" % (project_code, new_group)
             else:
                 parts = new_group.split("/")
                 title = parts[1]
+                user_defined_project_code = parts[0]
+                project_included = True
 
             new_group = Common.get_filesystem_name(new_group)
 
             group = SearchType.create("sthpw/login_group")
-            group.set_value("login_group", new_group)
-            group.set_value("project_code", project_code)
+
+            # the following sections contains the 4 cases possible when creating a new group
+            # through the popup
+            
+            # if the input is (project/name) and (not project specific)
+            if not is_project_specific and project_included:
+                
+                # Raise exception because it's contradictory data
+                raise TacticException('''You've given a project code while declaring the group not project specific. You can either not use the format "project_code/group_name", or check the "Project specific" checkbox.''')
+
+            # if the input is (project/name) and (project specific)
+            elif is_project_specific and project_included:
+
+                # first check is the user_defined_project_code is actually a project
+                # if it is, then set that as the project code. If not, don't set anything
+
+                expr = "@SOBJECT(sthpw/project['code', '%s'])" % user_defined_project_code
+                existing_project_codes = Search.eval(expr)
+
+                if existing_project_codes:
+                    group.set_value("project_code", user_defined_project_code)
+                else:
+                    raise TacticException('''The project code that you've given doesn't exist. Please choose an existing project code. Note: the format is "project_code/group_name". Projects include: %s''' % ", ".join(existing_project_codes))
```

Comment:
break it up into 2 lines or more less than 80 chars a line.. and since existing_project_codes is not really a code.. but a list of sobjects.. u can't do a join at the end.. Just don't display any existing project code.. at the end of this error message